cli/bin/semgrep

```diff
@@ -66,38 +72,42 @@ def find_semgrep_core_path():
+# we can always tell users to run pysemgrep instead of semgrep and be sure
+# they'll get the old behavior.
+def call_osemgrep():
+    path = find_semgrep_core_path()
+    # If you call semgrep-core as osemgrep, then we get osemgrep behavior,
+    # see src/main/Main.ml
+    sys.argv[0] = "osemgrep"
+    os.execvp(str(path), sys.argv)
+
+
+# When python imports a module, i.e. `import semgrep.FOO`, it runs the module's
 # top-level code, which is the code that is not inside any function or class.
-#
-# This means that if we try and run a function from the semgrep package in a subprocess,
-# we would end up calling os.execvp or main(), which causes a bunch of errors. So
-# we need to make sure that we only run the below code when this file is run as an executable.
-#
-# The above explanation is my understanding, but tbh I think python is weirder under the hood
-# with this stuff.
+# This means that if we try and run a function from the semgrep package in a
+# subprocess, we would end up calling os.execvp below or main(), which causes
+# a bunch of errors. So we need to make sure that we only run the code below
+# when this file is run as an executable, hence the guard below.
+# This fix a crash when using semgrep --test (which apparently triggers
+# execution of this file).
+# Austin: the above explanation is my understanding, but tbh I think Python
+# is weirder under the hood with this stuff.
 if __name__ == "__main__":
-    if "--experimental" in sys.argv:
-        #alt: we could  extend osemgrep (and pysemgrep) to accept a new --experimental
-        # flag instead of removing the flag below. However, in osemgrep we do a few
-        # pattern matching of Sys.argv and if we have 'osemgrep --experimental --help'
-        # this would be interpreted as 'osemgrep scan --experimental --help' (because
-        # scan is the default command), and we would not get the simple semgrep --help
-        # message showing all the subcommands. So simpler to just remove --experimental.
-        sys.argv.remove("--experimental")
-        # We could have moved the code below in a separate 'osemgrep', like for 'pysemgrep',
-        # but we don't want users to be exposed to another command, so better to hide it.
-        # We expose 'pysemgrep' because osemgrep itself might need to fallback to pysemgrep
-        # and it's better to avoid the possibility of an infinite loop by simply using
-        # a different program name. Morever, in case of big problems, we can always
-        # tell users to run pysemgrep instead of semgrep and be sure they'll get the
-        # old behavior.
-        path = find_semgrep_core_path()
-        # If you call semgrep-core as osemgrep, then we get osemgrep behavior,
-        # see src/main/Main.ml
-        sys.argv[0] = "osemgrep"
-        os.execvp(str(path), sys.argv)
+    if "--legacy" in sys.argv:
+       sys.argv.remove("--legacy")
+       #TODO? or execvp("pysemgrep", sys.argv)?
```

Comment:
No, we should not call execvp here, it will break the fix we just made for tests + autocomplete. I tried adding a longer explanation in that PR but the short answer is calling execvp from python to run more python breaks things

Reply:
I've added a bigger comment